src/main/groovy/nebula/plugin/dependencylock/utils/CoreLockingHelper.groovy

```diff
@@ -144,11 +144,44 @@ class CoreLockingHelper {
         additionalConfigNames
     }
 
-    private void removePreviousLockfiles() {
-        File gradleFilesDir = new File(project.projectDir, "gradle")
-        File lockfilesDir = new File(gradleFilesDir, "dependency-locks")
-        if (lockfilesDir.exists()) {
-            lockfilesDir.deleteDir()
+    private void removeLockfilesForUnlockedConfigurations() {
+        def migrationTaskWasRequested = project.gradle.startParameter.taskNames.contains(DependencyLockTaskConfigurer.MIGRATE_TO_CORE_LOCKS_TASK_NAME)
+        if (!shouldLockAllConfigurations && !migrationTaskWasRequested) {
+            project.gradle.taskGraph.whenReady { taskGraph ->
+                LinkedList tasks = taskGraph.executionPlan.executionQueue
+                Task lastTask = tasks.last?.task
+                taskGraph.addTaskExecutionListener(new TaskExecutionListener() {
+                    @Override
+                    void beforeExecute(Task task) {
+                        //DO NOTHING
+                    }
+
+                    @Override
+                    void afterExecute(Task task, TaskState taskState) {
+                        def thisTaskPathIsLastTaskPath = task.path == lastTask.path // should happen only once
+                        if (thisTaskPathIsLastTaskPath && !taskState.failure) {
+                            File gradleFilesDir = new File(project.projectDir, "gradle")
+                            File lockfilesDir = new File(gradleFilesDir, "dependency-locks")
+                            if (lockfilesDir.exists()) {
+                                def configNamesThatShouldBeLocked = configsWithActivatedDependencyLocking.collect {
+                                    "${it.name}.lockfile"
+                                }.toString()
+                                def shouldProvideInfoToLockAdditionalConfigurations = false
+                                lockfilesDir.listFiles().each { actualFile ->
+                                    if (!configNamesThatShouldBeLocked.contains(actualFile.name)) {
+                                        LOGGER.warn("Removing lockfile ${actualFile.name} as it is not configured for locking.")
```

Comment:
Warns the user that lockfiles are being removed
and later emits a message on how to lock those configurations